subseas_flex_fcst/maproom.py

```diff
@@ -45,26 +48,75 @@
 
 APP.layout = layout.app_layout
 
-
-def read_cptdataset(y_transform=False):
-
-    fcst_mu = cptio.open_cptdataset(Path(DATA_PATH, Path(CONFIG["forecast_mu_file"])))
+#function to open all datasets and then combine them
+def combine_cptdataset(dataDir,filePattern,dof=False,y_transform=False):
+    filesNameList = glob.glob(f'{dataDir}/{filePattern}')
+    dataDic = {}
+    for idx, i in enumerate(filesNameList):
+        fileCompList = re.split('[_.]',filesNameList[idx])
+        fileCompList = [x for x in fileCompList if "/" not in x]
+
+        leadTimeWk = [x for x in fileCompList if x.startswith('wk')]
+
+        dateFormat = re.compile(".*-.*-.*")
+        startDateStr = list(filter(dateFormat.match,fileCompList))
+        startDate = startDateStr[0]
+        #startDate = datetime.strptime(startDateStr[0],
+        #'%b-%d-%Y').date()
+        if leadTimeWk[0] == 'wk1':
+            leadTimeDate = 'Week 1'#'Apr 2' #should it be made into dates or kept as wk number?
+        elif leadTimeWk[0] == 'wk2':
+            leadTimeDate = 'Week 2'#'Apr 9'
+        elif leadTimeWk[0] == 'wk3':
+            leadTimeDate = 'Week 3'#'Apr 16'
+        elif leadTimeWk[0] == 'wk4':
+            leadTimeDate = 'Week 4'#'Apr 23'
+        leadTimeDict = {'L':[leadTimeDate]}
+
+        ds = cptio.open_cptdataset(filesNameList[idx])
+        if dof == True:
+            dofVar = float(ds["tp_pred_err_var"].attrs["dof"])
+            ds = ds.assign(dof=dofVar)
+            #dofDict = {'dof':[dof]}
+            #ds =ds.assign_coords(dof=("dof",[dof]))
+            #ds = ds.expand_dims(dofDict)
+        if len(ds['T']) == 1:
+            startDate
+            ds['T'] = [startDate] #for obs data there are many dates
+        ds = ds.expand_dims(leadTimeDict)
+        dataDic[idx] = ds
+    dataCombine = xr.combine_by_coords([dataDic[x] for x in dataDic],combine_attrs="drop_conflicts")
```

Comment:
Is drop_conflicts necessary? It seems like it could cover up mistakes that we would rather know about so we can fix them.

Reply:
Basically the issue was that, the dof (degree of freedom) was only expressed as a variable attribute within each data file, and each file might have a different dof associated with the data. So when I was combining the data sets, I was running into a problem because an attribute could not have more than one value. So this was the way I got around that (I do add the dof as a variable in the ds before they are removed as an attribute in line 77-79 to be used later)

Reply:
That sounds appropriate. I forgot this was about metadata attributes rather than data.